services/121-service/src/fsp-integrations/account-management/intersolve-visa-account-management/intersolve-visa-account-management.service.ts

```diff
@@ -317,17 +295,43 @@ export class IntersolveVisaAccountManagementService {
     return updatedWallet;
   }
 
-  /**
-   * Retrieves a registration by reference ID and program ID, and sends its contact information to Intersolve. Used only for debugging purposes.
-   */
   public async getRegistrationAndSendContactInformationToIntersolve(
     referenceId: string,
     programId: number,
   ): Promise<void> {
-    const registration = await this.getRegistrationOrThrow({
-      referenceId,
-      programId,
+    const registration = await this.registrationsService.getRegistrationOrThrow(
+      {
+        referenceId,
+        programId,
+      },
+    );
+
+    const dataFieldNames = getFspAttributeNames(Fsps.intersolveVisa);
+    const contactInformation: ContactInformation =
+      await this.registrationsService.getContactInformation({
+        referenceId,
+        programId,
+        dataFieldNames,
+      });
+
+    await this.intersolveVisaDataSynchronizationService.syncData({
+      registrationId: registration.id,
+      contactInformation,
     });
-    await this.sendCustomerInformationToIntersolve(registration);
+  }
+
+  private async throwIfCardDoesNotExistOrIsAlreadyLinked(
+    tokenCode: string,
+  ): Promise<void> {
+    // throws if tokenCode (card) does not exist
+    const intersolveVisaChildWallet =
+      await this.intersolveVisaService.getWallet(tokenCode);
+
+    if (intersolveVisaChildWallet.holderId) {
+      throw new HttpException(
+        `Card is already linked to another customer at Intersolve.`,
```

Comment:
```suggestion
        `Card is already linked to someone else`,
```

So this error is something that will actually appear in our frontend. I do not think that our users know the concept of 'customer at Intersolve'. So I think it's better to use something generic. 

I am no UX hero so I am open to better suggestions